src/main/java/net/dv8tion/jda/core/utils/WidgetUtil.java

```diff
@@ -0,0 +1,685 @@
+     */
+    public static String getPremadeWidgetHtml(Guild guild, WidgetTheme theme, int width, int height)
+    {
+        Args.notNull(guild, "Guild");
+        return getPremadeWidgetHtml(guild.getId(), theme, width, height);
+    }
+    
+    /**
+     * Gets the pre-made HTML Widget for the specified guild using the specified
+     * settings. The widget will only display correctly if the guild in question
+     * has the Widget enabled. Additionally, this method can be used independently
+     * of being on the guild in question.
+     * 
+     * @param guildId the guild ID
+     * @param theme the theme, light or dark
+     * @param width the width of the widget
+     * @param height the height of the widget
+     * @return a String containing the pre-made widget with the supplied settings
+     */
+    public static String getPremadeWidgetHtml(String guildId, WidgetTheme theme, int width, int height)
+    {
+        Args.notNull(guildId, "GuildId");
+        Args.notNull(theme, "WidgetTheme");
+        Args.notNegative(width, "Width");
+        Args.notNegative(height, "Height");
+        return String.format(WIDGET_HTML, guildId, theme.name().toLowerCase(), width, height);
+    }
+    
+    /**
+     * Represents the color scheme of the widget
+     * 
+     * These color themes match Discord's dark and light themes
+     */
+    public enum WidgetTheme
+    {
+        LIGHT, DARK
+    }
+    
+    
+    /**
+     * Makes a GET request to get the information for a Guild's widget. This
+     * widget (if available) contains information about the guild, including the
+     * Guild's name, an invite code (if set), a list of voice channels, and a
+     * list of online members (plus the voice states of any members in voice
+     * channels). This Widget can be obtained from any valid guild ID that has 
+     * it enabled; no accounts need to be on the server to access this information.
+     * 
+     * @param guildId
+     * @return null if the provided guild ID is not a valid Discord guild ID
+     * @return a Widget object with null fields and isAvailable() returning
+     *         false if the guild ID is valid but the guild in question does not
+     *         have the widget enabled
+     * @return a filled-in Widget object if the guild ID is valid and the guild
+     *         in question has the widget enabled.
+     */
+    public static Widget getWidget(String guildId)
+    {
+        Args.notNull(guildId, "GuildId");
+        // 400 - not valid snowflake
+        // 404 - guild not found
+        // 403 - widget disabled
+        // 200 - ok
+        try
+        {
+            HttpResponse<JsonNode> result = Unirest.get(String.format(WIDGET_URL, guildId)).asJson();
+            if (result.getStatus() == 403)
+                return new Widget(guildId);
+            if( result.getStatus() == 200)
+                return new Widget(result.getBody().getObject());
+            return null;
```

Comment:
This handling needs to be better. 429 is a possibility with this method. Don't just default to returning null. Deal with the codes you know and throw an exception detailing what went wrong + the code in the case of an unknown status.

Consider RateLimitException for 429